lib/matchers/matchers.ts

```diff
@@ -121,9 +121,15 @@ const solidity = (chai: any, utils: any) => {
       'Do not combine .not. with .withArgs()',
       expectedArgs.length,
       actualArgs.length);
-    for (let index = 0; index < expectedArgs.length; index++) {
-      new chai.Assertion(expectedArgs[index]).equal(actualArgs[index]);
-    }
+      for (let index = 0; index < expectedArgs.length; index++) {
+        if (expectedArgs[index].length !== undefined && typeof expectedArgs[index] !== "string"){
+            for (let j = 0; j < expectedArgs[index].length; j++){
+                new chai.Assertion(expectedArgs[index][j]).equal(actualArgs[index][j]);
+            }
+        } else {
+            new chai.Assertion((expectedArgs[index])).equal((actualArgs[index]));
+        }
+     }
```

Comment:
LGTM
Would you:

- add a test [here](https://github.com/EthWorks/Waffle/blob/master/test/matchers/events.ts) (will need to add example event [here](https://github.com/EthWorks/Waffle/blob/master/test/matchers/contracts/events.sol))
- perhaps extract method `assertArrayEqual(expectedArgs, actualArgs)`

Happy to merge and release after that.

Reply:
added tests and problematic event
passes eslint tests 
didn't extract the `assertArrayEqual` method for now